File: MyApi/controllers/AuthController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.IdentityModel.Tokens;
using MyApi.Dtos;
using MyApi.Models;
using MyApi.Services;
using System.IdentityModel.Tokens.Jwt;
using System.Security.Claims;
using System.Text;

namespace MyApi.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class AuthController : ControllerBase
    {
        private readonly UserService _userService;
        private readonly IConfiguration _config;

        public AuthController(UserService userService, IConfiguration config)
        {
            _userService = userService;
            _config = config;
        }

        [HttpPost("register-admin")]
        public async Task<IActionResult> RegisterAdmin([FromBody] RegisterDto dto)
        {
            var user = await _userService.CreateUser(dto.Name, dto.Email, dto.Password, Role.Admin);
            if (user == null) return BadRequest(new { message = "Admin already exists" });
            return Ok(new { message = "Admin registered", user = new { user.Id, user.Name, user.Email } });
        }

        [HttpPost("login")]
        public async Task<IActionResult> Login([FromBody] LoginDto dto)
        {
            var user = await _userService.Authenticate(dto.Email, dto.Password);
            if (user == null) return Unauthorized(new { message = "Invalid credentials" });
            if (!user.IsActive) return Forbid("Account deactivated");

            var tokenHandler = new JwtSecurityTokenHandler();
            var key = Encoding.ASCII.GetBytes(_config["Jwt:Key"]);

            var tokenDescriptor = new SecurityTokenDescriptor
            {
                Subject = new ClaimsIdentity(new[]
                {
                    new Claim("id", user.Id.ToString()),
                    new Claim(ClaimTypes.Role, user.Role.ToString())
                }),
                Expires = DateTime.UtcNow.AddDays(7),
                SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key), SecurityAlgorithms.HmacSha256Signature)
            };

            var token = tokenHandler.CreateToken(tokenDescriptor);
            return Ok(new
            {
                token = tokenHandler.WriteToken(token),
                user = new { user.Id, user.Name, user.Email, user.Role }
            });
        }
    }
}

// POST /api/auth/register-admin

// POST /api/auth/login

File: controllers/authController.js
import User from "../models/User.js";
import jwt from "jsonwebtoken";

const generateToken = (id, role) => {
  return jwt.sign({ id, role }, process.env.JWT_SECRET, { expiresIn: "7d" });
};

// Admin registers themselves
export const registerAdmin = async (req, res) => {
  const { name, email, password } = req.body;
  const exists = await User.findOne({ email });
  if (exists) return res.status(400).json({ message: "Admin already exists" });

  const admin = await User.create({ name, email, password, role: "admin" });
  res.status(201).json({ message: "Admin registered", admin });
};

// Login for admin or user
export const login = async (req, res) => {
  const { email, password } = req.body;
  const user = await User.findOne({ email });
  

  if (!user) return res.status(404).json({ message: "User not found" });
  if (!user.active)
    return res.status(403).json({ message: "Account deactivated" });

  

  const isMatch = await user.matchPassword(password);
  if (!isMatch) return res.status(401).json({ message: "Invalid credentials" });

  res.status(200).json({
    token: generateToken(user._id, user.role),
    user: { id: user._id, name: user.name, email: user.email, role: user.role },
  });
};
